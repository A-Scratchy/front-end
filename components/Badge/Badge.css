.Badge {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  margin: 1rem;
}

.iconContainer > img,
.iconContainer > svg {
  fill: currentColor;
  height: 50px;
  margin: 0.5rem 0;
}

.label {
  font-size: 1rem;
  line-height: 1rem;
  white-space: nowrap;
}
