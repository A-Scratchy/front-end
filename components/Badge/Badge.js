import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Badge.css';

Badge.propTypes = {
  svgComponent: PropTypes.element.isRequired,
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  labelClassName: PropTypes.string,
};

Badge.defaultProps = {
  className: undefined,
  labelClassName: undefined,
};

function Badge({ svgComponent, className, label, labelClassName }) {
  return (
    <div className={classNames(className, styles.Badge)}>
      <div className={styles.iconContainer}>{svgComponent}</div>
      <div className={classNames(styles.centerText, styles.label, labelClassName)}>{label}</div>
    </div>
  );
}

export default Badge;
