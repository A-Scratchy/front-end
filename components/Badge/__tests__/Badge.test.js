/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import Icon from 'static/images/icons/github_logo.svg';

import Badge from '../Badge';

const badgeIcon = <Icon />;

describe('Badge', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Badge svgComponent={badgeIcon} />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Badge svgComponent={badgeIcon} label="Badge Icon" className="test-class" />,
    );
  });
});
