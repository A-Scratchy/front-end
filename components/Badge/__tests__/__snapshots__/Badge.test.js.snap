// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render with many props assigned 1`] = `
<div
  className="Badge"
>
  <div
    className="iconContainer"
  >
    <test-file-mock />
  </div>
  <div
    className="centerText label"
  >
    Badge Icon
  </div>
</div>
`;

exports[`Badge should render with required props 1`] = `
<div
  className="Badge"
>
  <div
    className="iconContainer"
  >
    <test-file-mock />
  </div>
  <div
    className="centerText label"
  >
    
  </div>
</div>
`;
